<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PrefsDialog</class>
 <widget class="QDialog" name="PrefsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>396</width>
    <height>646</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>IDE Options</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>GroupBox</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0" colspan="2">
       <layout class="QFormLayout" name="formLayout_5">
        <item row="1" column="0">
         <widget class="QLabel" name="themeLabel">
          <property name="text">
           <string>UI Theme</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="themeWidget">
          <property name="toolTip">
           <string>Sets the UI theme</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="console1ColorLabel">
          <property name="text">
           <string>Console color#1</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="ColorSwatch" name="console1ColorWidget" native="true"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="console2ColorLabel">
          <property name="text">
           <string>Console color#2</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="ColorSwatch" name="console2ColorWidget" native="true"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="console3ColorLabel">
          <property name="text">
           <string>Console color#3</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="ColorSwatch" name="console3ColorWidget" native="true"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpOptions">
     <property name="title">
      <string>Code Editor Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <layout class="QFormLayout" name="formLayout_3">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="familyLabel">
          <property name="text">
           <string>Font Family</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QFontComboBox" name="fontComboBox"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="pointSizeLabel">
          <property name="text">
           <string>Font Size</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="fontSizeWidget">
          <property name="value">
           <number>12</number>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="tabSizeLabel_2">
          <property name="text">
           <string>Tab size</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="tabSizeWidget"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="highlightLabel">
          <property name="text">
           <string>Highlight:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QCheckBox" name="highlightCaretRowWidget">
          <property name="text">
           <string>Caret Row</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="smoothFontsLabel">
          <property name="text">
           <string>Smooth fonts</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QCheckBox" name="smoothFontsWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="showLineNumbersLabel">
          <property name="text">
           <string>Show line numbers</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QCheckBox" name="showLineNumbersWidget">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="sortCodeBrowserLabel">
          <property name="text">
           <string>Sort Code Browser</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QCheckBox" name="sortCodeBrowserWidget">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpColors">
     <property name="title">
      <string>Code Editor Colors</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="leftMargin">
       <number>9</number>
      </property>
      <property name="rightMargin">
       <number>9</number>
      </property>
      <item row="0" column="0">
       <layout class="QFormLayout" name="formLayout">
        <property name="fieldGrowthPolicy">
         <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="backgroundColorLabel">
          <property name="text">
           <string>Background</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="ColorSwatch" name="backgroundColorWidget" native="true"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="defaultColorLabel">
          <property name="text">
           <string>Default</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="ColorSwatch" name="defaultColorWidget" native="true">
          <property name="styleSheet">
           <string notr="true">color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="highlightColorLabel">
          <property name="text">
           <string>Highlight</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="ColorSwatch" name="highlightColorWidget" native="true"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="numbersColorLabel">
          <property name="text">
           <string>Numbers</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="ColorSwatch" name="numbersColorWidget" native="true"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="stringsColorLabel">
          <property name="text">
           <string>Strings</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="ColorSwatch" name="stringsColorWidget" native="true"/>
        </item>
        <item row="5" column="1">
         <widget class="ColorSwatch" name="identifiersColorWidget" native="true"/>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="keywordsColorLabel">
          <property name="text">
           <string>Keywords</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="ColorSwatch" name="keywordsColorWidget" native="true"/>
        </item>
        <item row="7" column="0">
         <widget class="QLabel" name="commentsColorLabel">
          <property name="text">
           <string>Comments</string>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="ColorSwatch" name="commentsColorWidget" native="true"/>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="identifiersColorLabel">
          <property name="text">
           <string>Identifiers</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="grpTools">
     <property name="title">
      <string>Tool Paths</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <layout class="QFormLayout" name="formLayout_4">
        <item row="0" column="0">
         <widget class="QLabel" name="cerberusLabel">
          <property name="text">
           <string>Cerberus</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="blitzMaxLabel">
          <property name="text">
           <string>BlitzMax</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QWidget" name="widget" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLineEdit" name="cerberusPathWidget">
             <property name="layoutDirection">
              <enum>Qt::LeftToRight</enum>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="cerberusPathButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QWidget" name="widget_2" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLineEdit" name="blitzmaxPathWidget">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="blitzmaxPathButton">
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="btnLoadColors">
       <property name="toolTip">
        <string>Load the editor colors from the theme</string>
       </property>
       <property name="text">
        <string>Load Colors</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnSaveColors">
       <property name="toolTip">
        <string>Save the editor colors with the theme</string>
       </property>
       <property name="text">
        <string>Save Colors</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="okayButton">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string>Okay</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorSwatch</class>
   <extends>QWidget</extends>
   <header>colorswatch.h</header>
   <container>1</container>
   <slots>
    <signal>colorChanged()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>themeWidget</tabstop>
  <tabstop>fontComboBox</tabstop>
  <tabstop>fontSizeWidget</tabstop>
  <tabstop>tabSizeWidget</tabstop>
  <tabstop>smoothFontsWidget</tabstop>
  <tabstop>highlightCaretRowWidget</tabstop>
  <tabstop>showLineNumbersWidget</tabstop>
  <tabstop>sortCodeBrowserWidget</tabstop>
  <tabstop>cerberusPathWidget</tabstop>
  <tabstop>cerberusPathButton</tabstop>
  <tabstop>blitzmaxPathWidget</tabstop>
  <tabstop>blitzmaxPathButton</tabstop>
  <tabstop>btnLoadColors</tabstop>
  <tabstop>btnSaveColors</tabstop>
  <tabstop>okayButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>stringsColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>379</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>keywordsColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>417</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>defaultColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>numbersColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>backgroundColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>identifiersColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>398</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>commentsColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>436</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontComboBox</sender>
   <signal>currentFontChanged(QFont)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onFontChanged(QFont)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fontSizeWidget</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onFontSizeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cerberusPathButton</sender>
   <signal>clicked()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onBrowseForPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>310</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blitzmaxPathButton</sender>
   <signal>clicked()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onBrowseForPath()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>522</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>395</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabSizeWidget</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onTabSizeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>highlightColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>143</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>smoothFontsWidget</sender>
   <signal>toggled(bool)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onSmoothFontsChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>176</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>okayButton</sender>
   <signal>clicked()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>570</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>283</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>highlightCaretRowWidget</sender>
   <signal>toggled(bool)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onHighlightCaretRowChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showLineNumbersWidget</sender>
   <signal>toggled(bool)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onShowLineNumbersChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>themeWidget</sender>
   <signal>currentTextChanged(QString)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onThemeChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>354</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sortCodeBrowserWidget</sender>
   <signal>toggled(bool)</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onSortCodeBrowserChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>249</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSaveColors</sender>
   <signal>clicked()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onSaveThemeColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>235</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLoadColors</sender>
   <signal>clicked()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onLoadThemeColor()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>43</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>493</x>
     <y>565</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>console1ColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>71</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>console2ColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>console3ColorWidget</sender>
   <signal>colorChanged()</signal>
   <receiver>PrefsDialog</receiver>
   <slot>onColorChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>322</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onColorChanged()</slot>
  <slot>onFontChanged(QFont)</slot>
  <slot>onFontSizeChanged(int)</slot>
  <slot>onBrowseForPath()</slot>
  <slot>onTabSizeChanged(int)</slot>
  <slot>onSmoothFontsChanged(bool)</slot>
  <slot>onHighlightCaretRowChanged(bool)</slot>
  <slot>onShowLineNumbersChanged(bool)</slot>
  <slot>onThemeChanged(QString)</slot>
  <slot>onSortCodeBrowserChanged(bool)</slot>
  <slot>onSaveThemeColor()</slot>
  <slot>onLoadThemeColor()</slot>
 </slots>
</ui>
